ccmlib/scylla_repository.py

```diff
@@ -69,22 +69,21 @@ def release_packages(s3_url, version, arch='x86_64', scylla_product='scylla'):
     :param version:
     :return:
     """
-    major_version = version if version.count('.') == 1 else '.'.join(n for n in version.split('.')[:2])
-    s3_url = s3_url.format(major_version)
 
+    major_version = extract_major_version(version)
+    s3_url = s3_url.format(major_version)
+    
     files_in_bucket = aws_bucket_ls(s3_url)
 
     if not files_in_bucket:
         raise RuntimeError(
             f"Failed to get release packages list for version {major_version}. URL: {s3_url}.")
 
-    all_unified_packages = [name for name in files_in_bucket if f'{scylla_product}-{arch}-unified-package' in name]
-    candidates = [package for package in all_unified_packages if ".rc" not in package and '.rc' not in version]
-
-    release_unified_packages = [package for package in candidates if version in package]
+    all_unified_packages = [name for name in files_in_bucket if f'{scylla_product}-{arch}-unified-package' in name or f'{scylla_product}-unified-package' in name]
+    release_unified_packages = [package for package in all_unified_packages if version in package]
 
     def extract_version(filename):
-        version_regex = re.compile(r'-(\d+!)?\d+([.-]\d+)?([.-]\d+)?([a-z]+\d*)?([.-](post|dev|rc)\d*)*')
+        version_regex = re.compile(r'-(\d+!)?\d+([.-]\d+)?([.-]\d+)?([a-z]+\d*)?([.~-](post|dev|rc)\d*)*')
```

Comment:
And here will be good to see names example, to know which templates are supported

Reply:
added examples